packages/docsearch-react/src/DocSearchButton.tsx

```diff
@@ -50,10 +50,10 @@ export const DocSearchButton = React.forwardRef<
       <span className="DocSearch-Button-Keys">
         {key !== null && (
           <>
-            <span className="DocSearch-Button-Key">
+            <kbd className="DocSearch-Button-Key">
               {key === ACTION_KEY_DEFAULT ? <ControlKeyIcon /> : key}
-            </span>
-            <span className="DocSearch-Button-Key">K</span>
+            </kbd>
+            <kbd className="DocSearch-Button-Key">K</kbd>
           </>
         )}
       </span>
```

Comment:
+1 for using semantic HTML. I'd also assume the check for `key` should happen one level higher as there's no point in rendering an empty span I'd imagine.


```tsx
      {key !== null && (
        <span className="DocSearch-Button-Keys">
          <kbd className="DocSearch-Button-Key">
            {key === ACTION_KEY_DEFAULT ? 'Search' : key}
          </kbd>
          <kbd className="DocSearch-Button-Key">K</kbd>
        </span>
      )}
```

Reply:
It was the case before, but we now use the first span to prevent UI flashes: https://github.com/algolia/docsearch/pull/1033